### PR TITLE
[DOC] Update tests for collapse search results

### PR DESCRIPTION
**Pull request by lockewritesdocs**
Updates tests for collapsing search results to use `desc` for sorting, and also fixes some other issues with remaining tests.

Resolves #74663

### CODE REVIEW

**Comment by lockewritesdocs** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by lockewritesdocs** [comment]
@nik9000, this is the PR where the [CI test](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+pull-request/job/elastic+elasticsearch+master+pull-request+docs-check/951/testReport/junit/org.elasticsearch.smoketest/DocsClientYamlTestSuiteIT/test__yaml_reference_search_search_your_data_collapse_search_results_line_191_/) is breaking. 

The bit that's failing is `max_score: null but was <missing>`, even though the output contains `null` as a value. I substituted a non-null value for `max_score` and tested locally with the same result. The error is then `max_score: <0.8630463> but was <missing>`.

File: docs/reference/search/search-your-data/collapse-search-results.asciidoc [hunk 1]
```diff
@@ -157,8 +184,25 @@ Second level of collapsing is also supported and is applied to `inner_hits`.
...
+----
+PUT my-index-000001/
+{"mappings":{"properties":{"@timestamp":{"type":"date"},"geo":{"properties":{"country_name":{"type":"keyword"}}},"http":{"properties":{"request":{"properties":{"method":{"type":"keyword"}}}}},"message":{"type":"text","fields":{"keyword":{"type":"keyword"}}},"user":{"properties":{"id":{"type":"keyword","doc_values":true}}}}}}
+----
+
+[source,console]
+----
+POST my-index-000001/_doc
+{"@timestamp":"2099-11-15T14:12:12","geo":{"country_name":"Amsterdam"},"http":{"request":{"method":"get"},"response":{"bytes":1070000,"status_code":200},"version":"1.1"},"message":"GET /search HTTP/1.1 200 1070000","source":{"ip":"127.0.0.1"},"user":{"id":"kimchy"}}
+----
```
Comment (nik9000) [inline docs/reference/search/search-your-data/collapse-search-results.asciidoc:200]: You need a refresh here!

**Review by nik9000 — COMMENTED** [review]
There is certainly something wrong with the error reporting code in mapmatcher - but I found a problem with the PR! I think it'll get you unstuck while I work on the error reporting.

**Pull request merged**